compiler/rustc_attr_parsing/src/context.rs

```diff
@@ -237,14 +239,22 @@ impl<'f, 'sess: 'f, S: Stage> AcceptContext<'f, 'sess, S> {
         })
     }
 
-    pub(crate) fn expected_no_args(&self, args_span: Span) -> ErrorGuaranteed {
-        self.emit_err(AttributeParseError {
-            span: args_span,
-            attr_span: self.attr_span,
-            template: self.template.clone(),
-            attribute: self.attr_path.clone(),
-            reason: AttributeParseErrorReason::ExpectedNoArgs,
-        })
+    pub(crate) fn expect_no_args(&self, args: &ArgParser<'_>) -> Result<(), ErrorGuaranteed> {
```

Comment:
I don't really like this method, it's different from all the other ones on here, and other parsers already solve this issue just inline. I don't really want the context to do any parsing really. I just want the methods to emit the errors, parsing happens in the parser.